spot-contracts/contracts/BondIssuer.sol

```diff
@@ -113,7 +124,35 @@ contract BondIssuer is IBondIssuer, OwnableUpgradeable {
 
     /// @inheritdoc IBondIssuer
     function isInstance(IBondController bond) external view override returns (bool) {
-        return _issuedBonds.contains(address(bond));
+        return (_activeBonds.contains(address(bond)) || _matureBonds.contains(address(bond)));
+    }
+
+    /// @inheritdoc IBondIssuer
+    /// @dev Reverts if none of the active bonds are mature.
+    function matureAll() external {
+        bool bondsMature = false;
+        uint256 activeCount_ = _activeBonds.length();
+
+        // NOTE: We traverse the active list in the reverse order as deletions involve
+        //       swapping the deleted element to the end of the list and removing the last element.
+        for (uint256 i = activeCount_; i > 0; i--) {
```

Comment:
Shouldn't this be...
```
for (uint256 i = activeCount_ - 1; i >= 0; i--) {
```

Edit: Ah, I see the subtraction on the next line.

A more idiomatic style is to cycle over the range in the storage container's index space, then use that normally in the code that follows.

This has a couple benefits:
- It reduces the chance of future bugs where someone uses `i` again but forgets to decrement it on each use.
-  If `i` is used more than once, it saves a small amount of gas.
- It's idiomatic, so future readers won't have their off-by-one alerts when they read it and it's easier to read quickly.

Reply:
^ This will throw an error when `activeCount_ = 0`